### PR TITLE
Add api endpoints

### PR DESCRIPTION
**Pull request by d4nQw3rty**
### This branch will implement the following feature:
**General requirements:**

- Make sure there are [no linter errors](https://github.com/microverseinc/linters-config).
- Make sure to use the correct flow [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
- Make sure to document work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).

**Ruby requirements**

- Follow Microverse's list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).

 **Project requirements:**

- Create an API endpoint to list all posts for a user.
image

- Create an API endpoint to list all comments for a user's post.
image

- Create an API endpoint to add a comment to a post. Remember that the owner of the comment is the user that makes it; it should not be set by passing a parameter.
image

- Your API endpoints should receive JSON and respond JSON as well.

### CODE REVIEW

File: app/controllers/api/v1/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+  # Use callbacks to share common setup or constraints between actions.
+  def set_comment
+    @comment = Comment.find(params[:id])
+  end
+
+  # Only allow a list of trusted parameters through.
+  def comment_params
+    params.require(:comment).permit(:id, :text)
+  end
```
Comment (MeqdamAlqudah) [inline app/controllers/api/v1/comments_controller.rb]: - [ ] Please getting the params from the user is not working like this always the user is getting this error when he tries to post on `api/v1/users/:user_id/posts/:post_id/comments` 
image
 
you can fix it by just permitting the parameter from  the params directly without the require(:comments) like this
image
you see it is working now

**Review by MeqdamAlqudah — CHANGES_REQUESTED** [review]
Hi @d4nQw3rty , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

image

 ## Highlights
:+1: Readme file 
:+1: no linter errors 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/api/v1/comments_controller.rb (hunk 1 above)
Comment (d4nQw3rty) [reply app/controllers/api/v1/comments_controller.rb]: Thanks for the review, @MeqdamAlqudah.  I will implement the change you are requiring.

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @d4nQw3rty 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉 

**Highlights**
- Well done making the requested changes from the previous reviewer
- Good communication

Cheers and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**